metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema_gen.py

```diff
@@ -1328,6 +1363,43 @@ def get_streams_for_schema(
 
         return streams.get(schema_name, [])
 
+    def fetch_procedures_for_schema(
+        self, snowflake_schema: SnowflakeSchema, db_name: str
+    ) -> List[BaseProcedure]:
+        try:
+            procedures: List[BaseProcedure] = []
+            for procedure in self.get_procedures_for_schema(snowflake_schema, db_name):
+                procedure_qualified_name = self.identifiers.get_dataset_identifier(
+                    procedure.name, snowflake_schema.name, db_name
+                )
+                self.report.report_entity_scanned(procedure_qualified_name, "procedure")
+
+                if self.filters.is_procedure_allowed(procedure_qualified_name):
+                    procedures.append(procedure)
+                else:
+                    self.report.report_dropped(procedure_qualified_name)
+            return procedures
+        except Exception as e:
+            if isinstance(e, SnowflakePermissionError):
+                error_msg = f"Failed to get procedures for schema {db_name}.{snowflake_schema.name}. Please check permissions."
+                raise SnowflakePermissionError(error_msg) from e.__cause__
+            else:
+                self.structured_reporter.warning(
+                    message="Failed to get procedures for schema",
+                    context=f"{db_name}.{snowflake_schema.name}",
+                    exc=e,
+                )
+                return []
```

Comment:
why raising exception if `SnowflakePermissionError` while gracefully warning and continue ingestion if not?

IMO graceful warning in both cases is the ideal scenario.

Same comment would apply to `fetch_streams_for_schema`

Reply:
Updated